src/editor/codemirror/linting/linter.ts

```diff
@@ -0,0 +1,37 @@
+import { Diagnostic, linter } from "@codemirror/lint";
+import { EditorView } from "@codemirror/view";
+// eslint-disable-next-line import/no-webpack-loader-syntax
+import Worker from "worker-loader!./linter.worker";
+
+const linterWorker = new Worker();
+
+export const tygerPythonLinter = () => {
+  const source = (view: EditorView): Promise<Diagnostic[]> => {
+    const code = view.state.doc.sliceString(0);
+    return new Promise((resolve) => {
+      linterWorker.addEventListener(
+        "message",
+        (e: MessageEvent) => {
+          const errors = e.data;
+          resolve(
+            errors.map((error: any) => {
+              return {
+                severity: "error",
+                // So that the go-to-next error feature works we highlight
+                // from the offset to the end of the line (as we don't have
+                // an end position). Not great when the error is at the end
+                // of a line.
```

Comment:
It's really quite bad at the end of the line and errors there are common (missing quote, colon, bracket etc).

Reply:
One for discussion rather than a quick tech fix I think. Moved to the PR summary.